addons/web/static/src/core/utils/strings.js

```diff
@@ -117,6 +117,35 @@ export function capitalize(s) {
     return s ? s[0].toUpperCase() + s.slice(1) : "";
 }
 
+/**
+ * Format the text as follow:
+ *      \*\*text\*\* => Put the text in bold.
+ *      --text-- => Put the text in muted.
+ *      \`text\` => Put the text in a rounded badge (bg-primary).
+ *      \<br> => Insert a breakline.
+ *      \<tab> => Insert 4 spaces.
+ *
+ * @param {string} text **will be escaped**
+ * @returns {string} string with the html that correspond to the wanted format
```

Comment:
I'll would make this function return the markuped string. I think it's more convenient for the code using it (which doesn't have to import both markup and odoomark), and it makes sense: we escape the string here, we introduce html tags => we wrap the result in a markup
```suggestion
 * @returns {Markup} string with the html that correspond to the wanted format
```